src/test/java/org/osiam/client/BugFilterWithNot.java

```diff
@@ -0,0 +1,94 @@
+@TestExecutionListeners({ DependencyInjectionTestExecutionListener.class,
+        DbUnitTestExecutionListener.class })
+@DatabaseSetup("/database_seeds/SearchUserServiceIT/database_seed_bug_filter_with_not.xml")
+@DatabaseTearDown(value = "/database_tear_down.xml", type = DatabaseOperation.DELETE_ALL)
+public class BugFilterWithNot extends AbstractIntegrationTestBase {
+    private static final String ADMIN_USER_NAME = "marissa";
+    private static final String ADMIN_USER_PASSWORD = "koala";
+    private static final String NAME_GROUP_TWO = "test_group02";
+    private final String TEST_USER_NAME = "marissa";
+    private final String TEST_USER_ID = "cef9452e-00a9-4cec-a086-d171374ffbef";
+
+    private AccessToken accessToken;
+
+    @Before
+    public void setup() {
+        accessToken = OSIAM_CONNECTOR.retrieveAccessToken(ADMIN_USER_NAME, ADMIN_USER_PASSWORD, Scope.ADMIN);
+    }
+
+    @Ignore("This test fails at the moment because the bug is not yet fixed. @Ignore must be removed after the bug-fix.")
+    @Test
+    public void filter_groups_where_the_user_is_not_included_in() throws Exception {
+        SCIMSearchResult<Group> result = searchGroup(String.format("not(members eq \"%s\")", TEST_USER_ID));
+
+        assertThat(String.format("Can not find expected group %s!", NAME_GROUP_TWO),
+                result.getResources(), hasItem(new Group.Builder(NAME_GROUP_TWO).build()));
+    }
+
+    @Ignore("This test fails at the moment because the bug is not yet fixed. @Ignore must be removed after the bug-fix.")
+    @Test
+    public void filter_users_where_not_a_member_of_group() throws Exception {
+        Group expectedGroup = getGroupByName(NAME_GROUP_TWO);
+
+        SCIMSearchResult<User> result = searchUser(String.format("not(groups eq \"%s\")", expectedGroup.getId()));
+
+        assertThat(String.format("The user %s is in the group %s but should not be!", TEST_USER_NAME, NAME_GROUP_TWO),
+                result.getResources(), not(hasItem(new User.Builder(TEST_USER_NAME).setId(TEST_USER_ID).build())));
+    }
+
+    private SCIMSearchResult<Group> searchGroup(String filter) {
+        QueryBuilder qb = new QueryBuilder();
+        qb.filter(filter);
+
+        return OSIAM_CONNECTOR.searchGroups(qb.build(), accessToken);
+    }
+
+    private SCIMSearchResult<User> searchUser(String filter) {
+        QueryBuilder qb = new QueryBuilder();
+        qb.filter(filter);
+
+        return OSIAM_CONNECTOR.searchUsers(qb.build(), accessToken);
+    }
+
+    private Group getGroupByName(String groupName) throws Exception {
+        Query query = new QueryBuilder().filter(String.format("displayName eq \"%s\"", groupName)).build();
+
+        SCIMSearchResult<Group> result = OSIAM_CONNECTOR.searchGroups(query, accessToken);
+        if (!result.getResources().isEmpty()) {
+            return result.getResources().get(0);
+        }
+
+        throw new Exception("The expected group was not found. This is an error with the test data.");
```

Comment:
Don't use conditions or `throw`s. The test will fail anyway if the database fixture could not be set up.